app/Http/Controllers/MainPlatform/ProfileController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Http\Controllers\MainPlatform;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\MainPlatform\ProfileUpdateRequest;
+use App\Models\User;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redirect;
+use Inertia\Inertia;
+use Inertia\Response;
+
+class ProfileController extends Controller
+{
+    /**
+     * Display the user's profile form.
+     */
+    public function edit(User $user): Response
+    {
+        return Inertia::render('Main-Platform/Profile/Edit', [
+            'user' => $user,
+            'status' => session('status'),
+        ]);
+    }
+
+    /**
+     * Update the user's profile information.
+     */
+    public function update(ProfileUpdateRequest $request): RedirectResponse
+    {
+        $request->user()->fill($request->validated());
+
+        if ($request->user()->isDirty('email')) {
+            $request->user()->email_verified_at = null;
+        }
+
+        $request->user()->save();
+
+        return Redirect::route('profile.edit');
+    }
+
+    /**
+     * Delete the user's account.
+     */
+    public function destroy(Request $request): RedirectResponse
+    {
+        $request->validate([
```

Comment:
Please use a form request for the validation to keep it consistent across the project.